File: the_basics/loops/for_loops/Python/for_loops.py
# Basic loop construction in Python

# Create some basic colletions
listOfNumbers = [1,2,3,4,5,6,7,8,9,10]
tupleOfCities = ("Chicago", "New York", "Paris", "London", "Tokyo")
collectionOfCars = {
    "Toyota" : "Corolla",
    "Subaru": "Forester",
    "Honda" : "Civic",
    "Jeep" : "Wrangler",
    "Ford" : "Mustang"
}
line_break = "\n"

# Note we do not have a basic for loop in python the way we do in other languages because the for (_initialization_; _condition_; _doThisForEachIteration) pattern does not work
# Instead we can only use for in loops.  In this case to match other for loop examples, I've included a range of numebrs for the iteration.

# -----------------BASIC LOOP-----------------------------------
for i in range(1,10):
    print(i) # <--  Output is numbers 1 through 10
print(line_break)

#-----------------COLLECTIONS LISTS TUPLES-----------------------

# Iterate through a list
for i in listOfNumbers:
    print(listOfNumbers[i - 1]) # <-- Output is numbers 1 through 10, notice we have to subtract 1 from i since list is zero indexed
print(line_break)

# Iterate through a tuple
for city in tupleOfCities:
    print(city) #<-- Output is all five cities listed in tuple
print(line_break)

#------------------COLLECTIONS DICTIONARIES ----------------------
for car in collectionOfCars:
    print(f"Make: {car}    Model: {collectionOfCars[car]}")
print(line_break)

File: the_basics/loops/for_loops/JavaScript/forLoops.js
// Basic for loop construction in JS

// Create some basic collections
var arrayOfNumbers = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];

var collectionOfCars = {
  Toyota: "Corolla",
  Subaru: "Forester",
  Honda: "Civic",
  Jeep: "Wrangler",
  Ford: "Mustang",
};

var lineBreak = "\n";

//-----------------BASIC LOOP-----------------------------------
// Iterate through a set of numbers
console.log("basic for loop");
for (i = 0; i < 10; i++) {
  console.log(i); // <-- Output will be all numbers from 1 to 9
}
console.log(lineBreak);

//-----------------COLLECTIONS ARRAYS --------------------------

console.log("array of numbers");
// Iterate through an array
for (i = 0; i < arrayOfNumbers.length; i++) {
  console.log(arrayOfNumbers[i]); // <--Output will be numbers 1 to 10
}
console.log(lineBreak);

// Using Built in Array method
console.log("array of numbers using forEach Array method");
arrayOfNumbers.forEach((number) => console.log(number)); //<--Output will also be 1 to 10
console.log(lineBreak);

//-------------------------COLLECTIONS OBJECTS ---------------------

// Iterate through a collection
console.log("collection of car make and models");
for (make in collectionOfCars) {
  console.log(`Make: ${make}   Model: ${collectionOfCars[make]}`); // <-- output will be each car i.e. "Make: Toyota   Model: Corolla"
}
